### PR TITLE
Extended tab auto complete to work with bundler and custom commands

### PR DESCRIPTION
**Pull request by revolter**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [ ] I've updated the documentation if necessary.

### Motivation and Context

This enabled tab auto complete for `bundle exec fastlane` and allows passing custom commands like `fl` to enable tab auto complete for `./fl`.

Works with:

    bundle exec fastlane enable_auto_complete
    bundle exec fastlane enable_auto_complete --custom first
    bundle exec fastlane enable_auto_complete --custom first,second

and:

    sh('bundle exec fastlane enable_auto_complete --custom first')

in a lane in the `Fastfile`.

### CODE REVIEW

File: fastlane/lib/fastlane/commands_generator.rb [hunk 1]
```diff
@@ -218,10 +218,11 @@ def run
       command :enable_auto_complete do |c|
         c.syntax = 'fastlane enable_auto_complete'
         c.description = 'Enable tab auto completion'
+        c.option '-c STRING[,STRING2]', '--custom STRING[,STRING2]', String, 'Add custom command(s) to use for auto complete'
```
Comment (milch) [inline fastlane/lib/fastlane/commands_generator.rb]: Could you expand a bit on this comment? From this description it sounds a bit unclear to me what this does

**Review by milch — COMMENTED** [review]
I like it, although I do think the possibility for it completing `bundle lane_name` could be a bit of an issue. Do we think that only advanced users will install shell completion anyway, so these users would not be confused by this? It looks like the auto completion for bundle is also added when calling `fastlane enable_auto_complete` without the `bundle exec`. Maybe we could work around any confusion by only adding the completion if it is called with `bundle exec`?

File: fastlane/lib/fastlane/commands_generator.rb (hunk 1 above)
Comment (revolter) [reply fastlane/lib/fastlane/commands_generator.rb]: > Add custom command(s) for which tab auto complete should work too

**Comment by revolter** [comment]
Indeed, installing completion generally means that you are accustomed to the terminal, and if you need to use the `bundle` command, chances that you know how it works are pretty high. In the worst case scenarios, `bundler` would print `Could not find command "deploy".` and `bundler: command not found: deploy`. Having no auto complete at all seems a worse solution, and there doesn't seem to be any way for it to work when typing the entire `bundle exec fastlane` string.

File: fastlane/lib/fastlane/commands_generator.rb (hunk 1 above)
Comment (milch) [reply fastlane/lib/fastlane/commands_generator.rb]: Yup, sounds better to me! Maybe `should be enabled too` instead of `should work too`

**Review by milch — APPROVED** [review]

**Pull request merged**